content/articles/how-to-send-an-email-with-django/index.md

```diff
@@ -0,0 +1,254 @@
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC"
+      crossorigin="anonymous"
+    />
+    <title>Send Email</title>
+  </head>
+  <body>
+    {% if messageSent %}
+    <div class="container">
+      <h3><strong>Email successfully sent</strong></h3>
+      <p>Your message was successfully sent to {{ form.cleaned_data.to }}</p>
+    </div>
+    {% else %}
+    <div class="container">
+      <h3><strong>Send E-mail</strong></h3>
+
+      <form method="post">
+        {% csrf_token %} {{ form.as_p }}
+
+        <br />
+        <div class="text-center">
+          <input
+            type="submit"
+            class="btn btn-secondary"
+            style="width: 100%"
+            value="Send e-mail"
+          />
+        </div>
+      </form>
+    </div>
+    {% endif %}
+  </body>
+</html>
+```
+
+#### What's happening here?
+1. We are checking if the variable `messageSent` is `True`. If it is, we show a success message.
+2. If it isn't true, we show an empty form for the users to fill out.
+
+We havenow successfully set up our project. In the following section, we will turn our attention to actual sending emails with Google Gmail SMTP provider.
+
+To test that all works well, update your `settings.py` file and add this configurations:
+
+```python
+EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
+
+DEFAULT_FROM_EMAIL = 'test@test.com'
+```
+
+Try submitting the form. The email will be printed out to your console. This is a good way of testing your application.
+
+### Configuring Google Gmail SMTP provider.
+To use your Google Gmail account to send an email, you will need to enable access for less secure applications, using [this link](https://myaccount.google.com/lesssecureapps).
+
+![Google Less Secure Homepage](/engineering-education/how-to-send email-with-django/google-less-secure.jpg)
+
+### Sending emails with Google Gmail SMTP provider
+To complete the process of actually sending emails using Google Gmail SMTP provider, you have to update and add some additional settings to your project with your Google Gmail account details!
+
+```python
+DEFAULT_FROM_EMAIL = '<paste your gmail account>'
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_HOST_USER = '<paste your gmail account'>'
+EMAIL_HOST_PASSWORD = '<paste the key or password app here>'
+EMAIL_PORT = 587
+EMAIL_USE_TLS = True
+```
+
+Note that some of the details like the account and password should be kept hidden. This can be achieved by setting them as environment variables. 
+
```

Comment:
Mentions methods or ways of hiding these credentials. Consider mentioning [Django-environ](https://django-environ.readthedocs.io/en/latest/#django-environ).